### PR TITLE
[14.0] FIX account_invoice_triple_discount - _recompute_tax_lines involving only relevant lines: invoice_line_ids

### PR DESCRIPTION
**Pull request by eLBati**
Ref https://github.com/OCA/l10n-italy/pull/3681

### CODE REVIEW

**Review by matteoopenf — APPROVED** [review]

**Review by odooNextev — APPROVED** [review]
LGTM

**Comment by francesco-ooops** [comment]
@eLBati can you add module name in PR title and commit name?

**Comment by eLBati** [comment]
> @eLBati can you add module name in PR title and commit name?

Done

**Comment by francesco-ooops** [comment]
cc @SirAionTech since you previously reviewed #1442

**Review by GSLabIt — APPROVED** [review]

**Comment by francesco-ooops** [comment]
@OCA/accounting-maintainers can this be merged please? thank you!

**Comment by francesco-ooops** [comment]
@tafaRU related to https://github.com/OCA/l10n-italy/pull/3681

**Review by sergiocorato — APPROVED** [review]
LGTM

**Comment by francesco-ooops** [comment]
@eLBati @sergiocorato this PR has broken tests for https://github.com/OCA/sale-workflow/tree/14.0/sale_triple_discount , see https://github.com/OCA/sale-workflow/actions/runs/6823495365/job/18557544202?pr=2771#step:8:1414

**Comment by francesco-ooops** [comment]
@SirAionTech in your opinion, does this need to be fw-ported to v16?

**Comment by SirAionTech** [comment]
> @SirAionTech in your opinion, does this need to be fw-ported to v16?

I haven't had time recently to look into this.
If I remember correctly from the long thread in https://github.com/OCA/account-invoicing/pull/1442, I was not very convinced of this change in the first place because I don't see how this is fixing this module.
So I have some concerns about back/forward-porting this.

Maybe a little recap of the history of this change might help understanding if it is needed in the future (`16.0`):
this change was originally needed for fixing the migration of **l10n_it_fatturapa_out_triple_discount** (https://github.com/OCA/l10n-italy/pull/3269).

It just happens that I have done the migration of the same module to `16.0`(https://github.com/OCA/l10n-italy/pull/3720); since this last PR is ok without this, I'd assume that this is not needed in `16.0`.

**TL;DR**: no

**Comment by francesco-ooops** [comment]
@sergiocorato @OCA/accounting-maintainers all PRs in sale-workflow v14 are failing due to this PR. 

Option 1) merge fix https://github.com/OCA/account-invoicing/pull/1605

Option 2) revert this change due to doubts expressed regarding to PR and fix in https://github.com/OCA/account-invoicing/pull/1592#issuecomment-1814530968, https://github.com/OCA/account-invoicing/pull/1605#discussion_r1395486680, https://github.com/OCA/account-invoicing/pull/1605#issuecomment-1814546796

Personally I find the second more favourable. This problem is not present in v16.
